pkg/util/log/log.go

```diff
@@ -74,7 +91,10 @@ func newPrometheusLogger(l logging.Level, format logging.Format, reg prometheus.
 
 // Log increments the appropriate Prometheus counter depending on the log level.
 func (pl *prometheusLogger) Log(kv ...interface{}) error {
+	start := time.Now()
 	pl.logger.Log(kv...)
+	pl.timeSpent.Observe(time.Since(start).Seconds())
```

Comment:
Not sure how much overhead this `Observe` is going to take on the hotpath logging. Hope it's worth it.

Reply:
That's a good point Kavi, I'm not actually sure.
This won't be around permanently, it's just designed so that we can measure the problem so when we implement the fix we can see if it helps or not.

In fact, I'll convert this to a draft so we can refer to it and cherry-pick it into a build, but I won't merge it

Reply:
If it is on the hot path, https://github.com/iovisor/bcc/blob/ee433d4e22650bc2ceca2119a727707f493269b3/tools/funclatency.py from eBPF is another option.

Reply:
Ooh that's interesting, something to try for sure if this doesn't work